src/include/common/file_system/file_system.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <filesystem>
+#include <vector>
+
+#include "file_info.h"
+
+namespace kuzu {
+namespace common {
+
+enum class FileLockType : uint8_t { NO_LOCK = 0, READ_LOCK = 1, WRITE_LOCK = 2 };
+
+class FileSystem {
+public:
+    virtual ~FileSystem() = default;
+
+    virtual std::unique_ptr<FileInfo> openFile(
+        const std::string& path, int flags, FileLockType lock_type = FileLockType::NO_LOCK) = 0;
+
+    virtual void readFromFile(
```

Comment:
As discussed, we should move functions that take in `fileInfo` to be non-public, as they will be duplicated with the interface inside `FileInfo`.